File: src/src_core/fmg/core/img/MosaicRotateTransformer.java
package fmg.core.img;

/** Representable {@link fmg.core.types.EMosaic} as animated image */
public class MosaicRotateTransformer implements IModelTransformer {

   @Override
   public void execute(IAnimatedModel model) {
      if (!(model instanceof MosaicAnimatedModel<?>))
         throw new RuntimeException("Illegal usage transformer");

      MosaicAnimatedModel<?> mam = (MosaicAnimatedModel<?>)model;

      double rotateAngleDelta = 360.0 / model.getTotalFrames(); // 360° / TotalFrames
//    //if (!mam.getAnimeDirection())
//    //   rotateAngleDelta = -rotateAngleDelta;
      double rotateAngle = mam.getCurrentFrame() * rotateAngleDelta;
      mam.setRotateAngle(rotateAngle);

      switch (mam.getRotateMode()) {
      case fullMatrix:
         mam.rotateMatrix();
         break;
      case someCells:
         mam.updateAnglesOffsets(rotateAngleDelta);
         mam.rotateCells();
         break;
      }
   }

}

File: src/src_core/fmg/core/img/MosaicRotateTransformer.cs
using System;

namespace Fmg.Core.Img {

    /// <summary> Representable <see cref="Fmg.Core.Types.EMosaic"/> as animated image </summary>
    public class MosaicRotateTransformer<TImage> : IModelTransformer
        where TImage : class
    {
        public void Execute(IAnimatedModel model) {
            if (!(model is MosaicAnimatedModel<TImage> mam))
                throw new InvalidOperationException("Illegal usage transformer");

            double rotateAngleDelta = 360.0 / mam.TotalFrames; // 360° / TotalFrames
            //if (!mam.AnimeDirection)
            //    rotateAngleDelta = -rotateAngleDelta;
            double rotateAngle = mam.CurrentFrame * rotateAngleDelta;
            mam.RotateAngle = rotateAngle;

            switch (mam.RotateMode) {
            case EMosaicRotateMode.fullMatrix:
                mam.RotateMatrix();
                break;
            case EMosaicRotateMode.someCells:
                mam.UpdateAnglesOffsets(rotateAngleDelta);
                mam.RotateCells();
                break;
            }
        }

    }

}
